app.py

```diff
@@ -0,0 +1,75 @@
+from bson.json_util import dumps
+from bson.objectid import ObjectId
+from flask import Flask, render_template, request, redirect, url_for
+from pymongo import MongoClient
+
+import datetime
+import json
+import os
+import sys
+
+client = MongoClient(os.environ['MONGODB_URI'])
+db = os.environ['MONGODB_URI'].split('/')[-1]
+app = Flask(__name__, static_url_path='')
+
+
+def get_5_last_documents():
+    return list(client[db].coll.find().sort("ts", -1).limit(5))
+
+
+@app.route('/add_document', methods=['POST'])
+def add_document():
+    error_msg = None
+
+    try:
+        document = json.loads(request.form['document'])
+        document["ts"] = str(datetime.datetime.now())
+    except Exception as e:
+        error_msg = "Converting to JSON failed: {}".format(e)
+
+    if not error_msg:
+        try:
+            resp = client[db].coll.insert(document)
+        except Exception as e:
+            error_msg = "Inserting into MongoDB failed: {}".format(e)
+
+    results = get_5_last_documents()
+
+    return redirect(url_for(
+        'index',
+        msg=error_msg if error_msg else "Document successfully added",
```

Comment:
This _could_ be just `error_msg or "string literal here"` - there are times where an explicit `x if y else z` avoids logic traps but I don't believe this is one of them. However, it's not quite as egregious as the below `False if var else True` which definitely scans as silly.

Reply:
that seems so javascript....